tests/python/ColorSpaceTest.py

```diff
@@ -1,31 +1,312 @@
+        for i, allocation in enumerate(OCIO.Allocation.__members__.values()):
+            self.assertEqual(allocation.name, OCIO.Allocation(i).name)
+            self.colorspace.setAllocation(allocation)
+            self.assertEqual(allocation, self.colorspace.getAllocation())
+
+        # Wrong type tests (using TransformDirection instead.)
+        for direction in OCIO.TransformDirection.__members__.values():
+            with self.assertRaises(TypeError):
+                self.colorspace.setAllocation(direction)
+
+        # Wrong type tests (set to None.)
+        with self.assertRaises(TypeError):
+            self.colorspace.setAllocation(None)
+
+    def test_allocation_vars(self):
+        """
+        Test the setAllocationVars() and getAllocationVars() methods.
+        """
+
+        # Array length tests
+        alloc_vars = []
+        for i in range(1, 5):
+            # This will create [0.1] up to [0.1, 0.2, 0.3, 0.4]
+            alloc_vars.append(float('0.%i' % i))
+            if i < 2 or i > 3:
+                with self.assertRaises(OCIO.Exception):
+                    self.colorspace.setAllocationVars(alloc_vars)
+            else:
+                self.colorspace.setAllocationVars(alloc_vars)
+                self.assertEqual(len(alloc_vars), len(
+                    self.colorspace.getAllocationVars()))
+
+        # Wrong type tests
+        wrong_alloc_vars = [['test'], 'test', 0.1, 1]
+        for wrong_alloc_var in wrong_alloc_vars:
+            with self.assertRaises(TypeError):
+                self.colorspace.setAllocationVars(wrong_alloc_var)
+
+    def test_bitdepth(self):
+        """
+        Test the setBitDepth() and getBitDepth() methods.
+        """
+
+        # Known constants tests
+        for i, bit_depth in enumerate(OCIO.BitDepth.__members__.values()):
+            self.assertEqual(bit_depth.name, OCIO.BitDepth(i).name)
+            self.colorspace.setBitDepth(bit_depth)
+            self.assertEqual(OCIO.BitDepth(bit_depth),
+                             self.colorspace.getBitDepth())
+
+        # Wrong type tests (using TransformDirection instead.)
+        for direction in OCIO.TransformDirection.__members__.values():
+            with self.assertRaises(TypeError):
+                self.colorspace.setBitDepth(direction)
+
+        # Wrong type tests (set to None.)
+        with self.assertRaises(TypeError):
+            self.colorspace.setBitDepth(None)
+
+    def test_category(self):
+        """
+        Test the hasCategory(), addCategory(), removeCategory(),
+        getCategories() and clearCategories() methods.
+        """
+
+        # Test empty categories
+        self.assertFalse(self.colorspace.hasCategory('ocio'))
+        self.assertEqual(len(self.colorspace.getCategories()), 0)
+        with self.assertRaises(IndexError):
+            self.colorspace.getCategories()[0]
+
+        # Test with defined TEST_CATS.
+        for i, y in enumerate(TEST_CATS):
```

Comment:
Recommend renaming `TEST_CATS` to `TEST_CATEGORIES`, only because `CAT` in color science means "Chromatic Adaptation Transform", and will have specific meaning in OCIO (`BuiltinTransform` uses it)